keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -200,17 +201,142 @@ def backbone_cls(cls):
     def preprocessor_cls(cls):
         return GPT2CausalLMPreprocessor
 
-    def _get_token_probability(self, prompt, mask):
+    def call_with_cache(self, token_ids, padding_mask, cache, cache_index=None):
+        """Forward pass of `GPT2CausalLM` with cache.
+
+        The difference between `call_with_cache` and normal `__call__` is in
+        this method, a `cache` arg is set, and the inputs is of
+        `sequence_length=1`. By cachine the previous key/value in multi-head
+        attention, we avoid recomputing the outputs of seen tokens.
+
+        Args:
+            token_ids: a dense int Tensor, input token ids.
+            padding_mask: a dense bool Tensor, input padding mask.
+            cache: a dense float Tensor, the cache of key and value.
+            cache_index: int, or int Tensor, defaults to None. If set, it
+                represents the index of current inputs in the whole sequence.
+
+        Returns:
+            x: a dense float Tensor, the next token logits of `inputs`.
```

Comment:
Same issue here as with TransformerDecoder. Rewrite this to reference the tuple structure, the name of the tuple outputs in our implementation should be totally irrelevant. E.g.

A `(logits, cache)` tuple. Where the first output is the language model logits for the input `token_ids` and the second output is the updated cache.